src/scanner.rs

```diff
@@ -0,0 +1,262 @@
+use crate::error::error;
+use crate::token::{Token, TokenType};
+use crate::util::*;
+
+use std::iter::Peekable;
+use unicode_segmentation::{Graphemes, UnicodeSegmentation};
```

Comment:
interesting that you go by graphemes

tho ig it makes sense, especially for line/column stuff